db/migrations/20200603124229_create_time_ilk_snapshot_query.sql

```diff
@@ -0,0 +1,66 @@
+-- +goose Up
+CREATE TYPE api.time_ilk_snapshot AS
+(
+    bucket_start TIMESTAMP,
+    bucket_end TIMESTAMP,
+    bucket_interval INTERVAL,
+    ilk_identifier TEXT,
+    block_number BIGINT,
+    rate NUMERIC,
+    art NUMERIC,
+    spot NUMERIC,
+    line NUMERIC,
+    dust NUMERIC,
+    chop NUMERIC,
+    lump NUMERIC,
+    flip TEXT,
+    rho NUMERIC,
+    duty NUMERIC,
+    pip TEXT,
+    mat NUMERIC,
+    created TIMESTAMP,
+    updated TIMESTAMP
+);
+
+CREATE FUNCTION api.time_ilk_snapshots(ilk_identifier TEXT, bucket_start TIMESTAMP, bucket_end TIMESTAMP, bucket_interval INTERVAL)
```

Comment:
Are there any inputs that could cause this function to fail or perform in undesirable ways? e.g. if I pass in a `bucket_start` that's greater than `bucket_end`, or pass in a `bucket_interval` that's very small?

Reply:
I just ran some tests. If `range_start` (formerly `bucket_start`) is greater than `range_end` it will just return 0 rows. This seems like the intuitive result to me.

I also tested a very small `bucket_interval` of 1 minute over a day. Other than just taking a bit of time to process, it returns the results as expected with a lot of duplicate rows if there are no changes for several blocks.

Reply:
Marking this as resolved. Re-open if this is still something to discuss.

Reply:
The taking a bit of time part is a little bit concerning to me, if arbitrary users can input very large ranges and and very small intervals. Do you think there are any range + interval combinations that we could enable, disabling all others, to support the most common use cases? E.g. I could see enabling a version of this query that _only_ supports, say, the last 100 days with 1 day intervals, or the last week with 1 hour intervals, etc.